File: frontend/bff/src/application/organizations/commands/add-member.command.ts
import type { Desk } from '../../../cross-cutting/enums/desk.js';
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface AddMemberCommand {
  context: RequestContext;
  email: string;
  /** Alçada inicial (opcional): código de uma alçada da organização. */
  ruleCode: string | null;
  desk: Desk | null;
}

File: backend/core-service/src/Fix.Application/UseCases/Organizations/Commands/AddMemberCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Domain.AggregateRoots.Organizations;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Organizations.Commands;

/// <summary>
/// Adiciona um membro como user (na organização FIX, como administrador interno). RuleCode opcional: código de uma
/// alçada personalizada da organização a atribuir já na entrada (vazio ou "user" = sem alçada).
/// </summary>
[RequireRole(RoleCodes.CreateUser)]
public sealed record AddMemberCommand(Guid UserId, Guid OrganizationId, string Email, string? RuleCode, Desk? Desk)
    : ICommand<Guid>, IOrganizationRequest;

